code/game/objects/items/devices/uplinks.dm

```diff
@@ -69,6 +74,9 @@ var/list/world_uplinks = list()
 			if(I.job && I.job.len)
 				if(!(I.job.Find(job)))
 					continue
+			if(I.species && I.species.len)
+				if(!(I.species.Find(species)))
```

Comment:
have this be iterated on per-object basis so you can check for species with claws by flag instead